target/scripts/startup/daemons-stack.sh

```diff
@@ -1,160 +1,86 @@
-function _start_daemons_saslauthd
+function _default_start_daemon
 {
-  _log 'debug' 'Starting saslauthd'
-  supervisorctl start "saslauthd_${SASLAUTHD_MECHANISMS}" || dms_panic__fail_init 'saslauthd'
-}
+  _log 'debug' "Starting ${1:?}"
 
-function _start_daemons_fail2ban
-{
-  _log 'debug' 'Starting Fail2ban'
-  touch /var/log/auth.log
+  local RESULT
+  RESULT="$(supervisorctl start "${1}" 2>&1)"
 
-  # delete fail2ban.sock that probably was left here after container restart
-  if [[ -e /var/run/fail2ban/fail2ban.sock ]]
+  # shellcheck disable=SC2181
+  if [[ ${?} -ne 0 ]]
   then
-    rm /var/run/fail2ban/fail2ban.sock
+    echo "${RESULT}" >&2
+    dms_panic__fail_init "${1}"
   fi
-
-  supervisorctl start fail2ban || dms_panic__fail_init 'Fail2ban'
 }
 
-function _start_daemons_opendkim
-{
-  _log 'debug' 'Starting opendkim'
-  supervisorctl start opendkim || dms_panic__fail_init 'opendkim'
-}
+function _start_daemon_changedetector { _default_start_daemon 'changedetector' ; }
+function _start_daemon_amavis         { _default_start_daemon 'amavis'         ; }
+function _start_daemon_clamav         { _default_start_daemon 'clamav'         ; }
+function _start_daemon_cron           { _default_start_daemon 'cron'           ; }
+function _start_daemon_opendkim       { _default_start_daemon 'opendkim'       ; }
+function _start_daemon_opendmarc      { _default_start_daemon 'opendmarc'      ; }
+function _start_daemon_postsrsd       { _default_start_daemon 'postsrsd'       ; }
+function _start_daemon_postfix        { _default_start_daemon 'postfix'        ; }
+function _start_daemon_rsyslog        { _default_start_daemon 'rsyslog'        ; }
+function _start_daemon_update_check   { _default_start_daemon 'update-check'   ; }
 
-function _start_daemons_opendmarc
+function _start_daemon_saslauthd
 {
-  _log 'debug' 'Starting opendmarc'
-  supervisorctl start opendmarc || dms_panic__fail_init 'opendmarc'
+  _default_start_daemon "saslauthd_${SASLAUTHD_MECHANISMS}"
 }
 
-function _start_daemons_postsrsd
+function _start_daemon_postgrey
 {
-  _log 'debug' 'Starting postsrsd'
-  supervisorctl start postsrsd || dms_panic__fail_init 'postsrsd'
+  rm -f /var/run/postgrey/postgrey.pid
+  _default_start_daemon 'postgrey'
 }
 
-function _start_daemons_postfix
+function _start_daemon_fail2ban
 {
-  _log 'debug' 'Starting postfix'
-  supervisorctl start postfix || dms_panic__fail_init 'postfix'
+  touch /var/log/auth.log
+
+  # delete fail2ban.sock that probably was left here after container restart
+  [[ -e /var/run/fail2ban/fail2ban.sock ]] && rm /var/run/fail2ban/fail2ban.sock
+
+  _default_start_daemon 'fail2ban'
 }
 
-function _start_daemons_dovecot
+function _start_daemon_dovecot
 {
-  _log 'debug' 'Starting dovecot services'
-
   if [[ ${ENABLE_POP3} -eq 1 ]]
   then
-    _log 'debug' 'Starting pop3 services'
-    mv /etc/dovecot/protocols.d/pop3d.protocol.disab \
-      /etc/dovecot/protocols.d/pop3d.protocol
+    _log 'debug' 'Starting POP3 services'
```

Comment:
Technically it's enabling it through config for Dovecot to manage?

```suggestion
    _log 'debug' 'Enabling POP3 services'
```